clients/web/src/presentation/pages/auth/Login/useLogin.ts

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { getUser, userSignIn } from '@client-commons/platform/aws/auth/awsAuth'
+import useFetchData from '@client-commons/hooks/useFetchData'
+import { DashboardPath } from '../../../../constants/routeConsts'
+
+type UseLoginReturn = {
+  email: string;
+  setEmail: React.Dispatch<React.SetStateAction<string>>;
+  password: string;
+  setPassword: React.Dispatch<React.SetStateAction<string>>;
+  toastVisible: boolean;
+  toastMsg: string;
+  toastClass: string;
+  loading: boolean;
+  showPassword: boolean;
+  handleTogglePasswordVisibility: () => void;
+  handleLogin: () => Promise<void>;
+}
+
+export const useLogin = (): UseLoginReturn => {
+  const navigate = useNavigate()
+  const [email, setEmail] = useState<string>('')
+  const [password, setPassword] = useState<string>('')
+  const [toastVisible, setToastVisible] = useState<boolean>(false)
+  const [toastMsg, setToastMsg] = useState<string>('')
+  const [toastClass, setToastClass] = useState<string>('')
+  const [showPassword, setShowPassword] = useState<boolean>(false)
+
+  const [signIn, loading, , signInError] = useFetchData(userSignIn)
+  const [fetchUser, , user] = useFetchData(getUser)
+
+  useEffect(() => {
+    void fetchUser()
+  }, [fetchUser])
+
+  useEffect(() => {
+    if (user != null) {
```

Comment:
i feel like you should also check for error?
if user is not null and error is null. Since we might have a state that the user is available but a refresh auth token has errored out?